packages/compass-connections/src/hooks/use-connection-repository.ts

```diff
@@ -0,0 +1,131 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import ConnectionString from 'mongodb-connection-string-url';
+import { merge } from 'lodash';
+import isEqual from 'lodash/isEqual';
+import { ConnectionStorageEvents } from '@mongodb-js/connection-storage/renderer';
+import { useConnectionStorageContext } from '@mongodb-js/connection-storage/provider';
+import { useState, useEffect, useCallback } from 'react';
+import { BSON } from 'bson';
+
+type DeepPartial<T> = T extends object
+  ? { [P in keyof T]?: DeepPartial<T[P]> }
+  : T;
+
+export type PartialConnectionInfo = DeepPartial<ConnectionInfo> &
+  Pick<ConnectionInfo, 'id'>;
+
+/**
+ * Same as _.isEqual, except it takes key order into account
```

Comment:
I think this was copied verbatim from the other place we use a similar check, but in this case I guess we care more about being able to compare bson values, not particularly the key order, maybe worth updating.

Reply:
Good point. I'll change the comment. This check was in useActiveConnections, and I moved it here as useActiveConnections will use this hook, so it make sense to move this utility here so only useActiveConnections depend on useConnectionRepository and not the other way around with a cyclic dependency.